modules/core/include/opencv2/core/operations.hpp

```diff
@@ -210,8 +210,8 @@ Matx<_Tp, n, l> Matx<_Tp, m, n>::solve(const Matx<_Tp, m, l>& rhs, int method) c
 {
     Matx<_Tp, n, l> x;
     bool ok;
-    if( method == DECOMP_LU || method == DECOMP_CHOLESKY )
-        ok = cv::internal::Matx_FastSolveOp<_Tp, m, l>()(*this, rhs, x, method);
+    if( m == n && (method == DECOMP_LU || method == DECOMP_CHOLESKY) )
```

Comment:
I think this does not solve the problem and it will cause the same compilation error.

Reply:
At compile time, the compiler won't check the conditions inside `if`.